libxmf/XmfVpxPacket.h

```diff
@@ -0,0 +1,46 @@
+#ifndef XMF_VPX_PACKET_H
+#define XMF_VPX_PACKET_H
+
+#include <vpx/vpx_codec.h>
+#include <xmf/xmf.h>
+#include <vpx/vpx_encoder.h>
+#include "XmfVpxFrame.h"
+
+/// <div rustbindgen opaque></div>
+typedef struct xmf_vpx_packet XmfVpxPacket;
+
+typedef enum xmf_vpx_packet_kind
+{
+    XMF_VPX_CODEC_CX_FRAME_PKT = VPX_CODEC_CX_FRAME_PKT,     /**< Compressed video frame */
+    XMF_VPX_CODEC_STATS_PKT = VPX_CODEC_STATS_PKT,           /**< Two-pass statistics for this frame */
+    XMF_VPX_CODEC_FPMB_STATS_PKT = VPX_CODEC_FPMB_STATS_PKT, /**< first pass mb statistics for this frame */
+    XMF_VPX_CODEC_PSNR_PKT = VPX_CODEC_PSNR_PKT,             /**< PSNR statistics for this frame */
+    XMF_VPX_CODEC_CUSTOM_PKT = 256                           /**< Algorithm extensions  */
+} XmfVpxPacketKind;
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+    XMF_EXPORT void XmfVpxPacket_Destroy(XmfVpxPacket *packet);
+    XMF_EXPORT XmfVpxPacketKind XmfVpxPacket_GetKind(const XmfVpxPacket *packet);
+    /**
+     * Retrieves the frame data from the packet.
+     * This function will return NULL if the packet is not a frame packet.
+     * @param packet Pointer to the packet instance.
+     * @return Pointer to the frame data, or NULL if the packet is not a frame packet.
+     *
+     * @note The returned frame data is owned by the packet and should not be freed by the caller.
+     */
+    XMF_EXPORT XmfVpxFrame *XmfVpxPacket_GetFrame(const XmfVpxPacket *packet);
```

Comment:
issue: `@note The returned frame data is owned by the packet and should not be freed by the caller.`

If this is true, then we have a problem here:

```rust
        // SAFETY: see SAFETY Note in VpxPacket
        let frame_ptr = unsafe { XmfVpxPacket_GetFrame(self.ptr) };
        if frame_ptr.is_null() {
            None
        } else {
            // SAFETY: The frame pointer is not null and is valid, since the packet is valid. XmfVpxFrame is deep copied on creating from XmfVpxPacket.
            Some(unsafe { VpxFrame::from_raw(frame_ptr) })
        }
```

Verify whether `VpxFrame` should borrow the packet data instead, or update the doc of `XmfVpxPacket_GetFrame` to match the actual behavior.

Reply:
I was an old comment from my previous experiment, now it's indeed a deep copy of the frame data, updated comment

Reply:
updated